pkg/remote/aws/repository/ec2_repository.go

```diff
@@ -328,3 +335,30 @@ func (r *ec2Repository) ListAllSecurityGroups() ([]*ec2.SecurityGroup, []*ec2.Se
 	r.cache.Put("ec2ListAllDefaultSecurityGroups", defaultSecurityGroups)
 	return securityGroups, defaultSecurityGroups, nil
 }
+
+func (r *ec2Repository) ListAllNetworkACLs() ([]*ec2.NetworkAcl, error) {
+
+	cacheKey := "ec2ListAllNetworkACLs"
+	r.locks[cacheKey].Lock()
+	defer r.locks[cacheKey].Unlock()
```

Comment:
not sure if it's a good idea to implement an independent solution for #963 in this PR, I think we need to agree on a common solution for all repositories and providers. WDYT?

Reply:
> From this lock, I understand that the lock inside the r.cache.Get is not enough if the method ListAllNetworkACLs gets called twice in our ParallelRunner. Is that the case?

No, since if we got a cache miss we only want to release the lock after we read value from AWS. If we do not do that you can have two parallel cache miss, we have opened an issue for that : #963 
 
> not sure if it's a good idea to implement an independent solution for #963 in this PR, I think we need to agree on a common solution for all repositories and providers. WDYT?

The idea was not to implement a solution for #963 here but only a single required lock for this repository. Do you think that we should ship this PR without any lock knowing that it will be broken ?

Reply:
> Do you think that we should ship this PR without any lock knowing that it will be broken ?

We already have tons of resources that misses a locking system anyway, may be #963 should be considered as a release blocker then?

Reply:
I'm not against fixing this in newer resources but let's be clear right now before merging new ones.

Reply:
> There is a difference between refactoring legacy code to fix technical debt and shipping new code with known technical debt

I agree with that, I just think it's better to solve the problem once for all. May be we could ensure the method locks as expected in tests for new resources, so we avoid any regression when we will work on #963 ?

Reply:
I wonder if we could have a solution where the cache contains the lock and we call it the same way from all the repository.
something like 
```go
val, lock := r.cache.Read(cacheKey);
if  val != nil {
   return val.([]*ec2.NetworkAcl), nil
}
defer lock..Unlock()
```
And cache would lock if there is a miss and pass the responsability for the unlock ?

Reply:
We can debate about implementation details later. The main question for this PR is : should we handle lock in future repositories or should we wait for #963 to fix them all in a second time ?

Reply:
If we want a common solution for all I think we should not fix it in new resources I think.
But if we want each repo to handle it's own lock (which makes sense) we could fix in new resources and take care of the old ones later

Reply:
I removed locks for now, this PR is big enough, we'll see how to fix this in #963. Maybe we should prioritize it.